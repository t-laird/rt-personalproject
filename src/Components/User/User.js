import React, { Component } from 'react';
import { connect } from 'react-redux';
import UserData from '../UserData/UserData';
import UserProfile from '../UserProfile/UserProfile';
import * as actions from '../../Actions/';
import './User.css';
import { NavLink } from 'react-router-dom';
import Transaction from '../Transaction/Transaction';
import Comments from '../Comments/Comments';

export class User extends Component {
  constructor(props) {
    super(props);

    this.state = {
      joinText: 'JOIN A GROUP'
    };
  }

  componentDidMount() {
    if (!this.props.User.name) {
      this.props.history.push('/login');
    }
    if (this.props.User.group_id > 0) {
      this.setState({joinText: 'SWITCH GROUPS'});
    }
  }

  render() {
    return (
      <div className="user-component">
        <UserData />
        <Transaction />
        <Comments />
        <UserProfile />
        <NavLink className="join-group" to='/joingroup'>{this.state.joinText}</NavLink>
      </div>
    );
  }
}

export const mapStateToProps = ( store ) => ({
  User: store.User
});


export default connect(mapStateToProps, null)(User);