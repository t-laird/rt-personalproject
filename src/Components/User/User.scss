@import '../../Variables.scss';

.user-component {
  padding: 3px 3px 50px;
  width: 100%;
  @include component-transition;

  .join-group {
  	width: 750px;
  	margin: auto;
  	display: block;
  	padding: 0 25px;
  	color: $c1;
  	transition: color .15s ease-in-out;
  	font-weight: 900;
  	text-align: right;
    font-size: 20px;
  }

  .join-group:hover {
  	color: darken($c1, 12%);
  }
}

.user-component.fade-enter {
  @include component-entering;
}

.user-component.fade-exit {
  @include component-exiting;
}

@media screen and (max-width: 750px) {
  .user-component {
    padding: 0 0 50px;

    .join-group {
      width: 100%;
    }
  }
}
