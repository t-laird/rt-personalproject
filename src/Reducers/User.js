const User = (state = {}, action) => {
  switch(action.type) {
    case "UPDATE_USER":
      return action.user

    case "LOGOUT_USER":
    	return state = {}

    default:
      return state;
  }
}

export default User;