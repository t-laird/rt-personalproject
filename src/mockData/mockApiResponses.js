export const getGroupSettingsApiResponse = {
  status: 'success',
};

export const getGroupTransactionDataApiResponse = {
	status: 'success'
};