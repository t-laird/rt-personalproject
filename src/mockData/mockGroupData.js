export const mockGroupData = {
	administrator_id: 95,
	created_date: "2017-10-18T18:34:30.017Z",
	group_id: 1,
	group_name: 'Turing',
	group_passphrase: "9139z",
	weekly_points: 100
}