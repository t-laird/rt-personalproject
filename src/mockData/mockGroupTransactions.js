const mockGroupTransactions = [
  {transactions: [
    {
      created_time: 1514939019586,
      event_id: 9,
      event_time: "2018-01-03T00:23:39.627Z",
      group_id: 1,
      note: "You totally killed it on your lightning talk this morning!  I loved it.",
      point_value: 15,
      receive_id: 22,
      received_name: "Scott Adams",
      send_id: 24,
      send_name: "Jim Davis"
    },
    {
      created_time: 1514939019586,
      event_id: 2,
      event_time: "2018-01-03T00:23:39.612Z",
      group_id: 1,
      note: "Thanks!",
      point_value: 10,
      receive_id: 59,
      received_name: "Bo Jackson",
      send_id: 43,
      send_name: "Deion Sanders"
    }
  ]}
]

export default mockGroupTransactions;