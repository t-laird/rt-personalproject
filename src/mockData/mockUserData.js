export const mockUserData = {
  authrocket_id: "usr_0vXWX9jSRkbUjLGsOb9S8r",
  created_date: "2017-11-18T18:34:30.017Z",
  email: "cpavin@usopen.com",
  group_id: 1,
  name: "Corey Pavin",
  user_id: 95
};

