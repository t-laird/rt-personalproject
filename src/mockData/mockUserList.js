export const mockUserList = [
  {
    authrocket_id: "jfnxcio42393",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "abcf@gmail.com",
    group_id: 1,
    name: "Mike Jenkins",
    user_id: 2
  },
  {
    authrocket_id: "lksdfjn23413",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "qwi123@gmail.com",
    group_id: 1,
    name: "Josh Brown",
    user_id: 3
  },
  {
    authrocket_id: "xmcv3941nv",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "293xcnv@gmail.com",
    group_id: 1,
    name: "Alan Michaelson",
    user_id: 4
  },
  {
    authrocket_id: "3490xxa1j",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "nc-1n23a@gmail.com",
    group_id: 1,
    name: "Sarah Hilman",
    user_id: 5    
  } 
];