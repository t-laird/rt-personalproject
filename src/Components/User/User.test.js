/* eslint-disable max-len */

import React from 'react';
import { User, mapStateToProps } from './User';
import { shallow } from 'enzyme';
import { mockUserData } from '../../mockData/mockUserData';
import { mockUserWithoutGroup } from '../../mockData/mockUserWithoutGroup';

describe('user tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<User User={mockUserData} history={[]} />);

    expect(renderedApp).toMatchSnapshot();
  });

  it('should match the snapshot if user does not have a group', () => {
    const renderedApp = shallow(<User User={mockUserWithoutGroup} history={[]} />);
  
    expect(renderedApp).toMatchSnapshot();
  });

  it('should auto-redirect if no user is logged in', () => {
    const renderedApp = shallow(<User User={{}} history={[]} />);

    expect(renderedApp.instance().props.history[0]).toEqual('/snap-ninja/login');
  });
  
  it('should not redirect if a user is logged in', () => {
    const renderedApp = shallow(<User User={mockUserData} history={[]} />);

    expect(renderedApp.instance().props.history.length).toEqual(0);
  });
});

describe('mapStateToProps tests', () => {
  it('should pull user from store', () => {
    const mockStore = { mockUserData };
    const result = mapStateToProps(mockStore);

    expect(result.UserData).toEqual(mockStore.UserData);
  });
});