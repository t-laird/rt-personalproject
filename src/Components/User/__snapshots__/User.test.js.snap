// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <User
    User={
        Object {
            "authrocket_id": "usr_0vXWX9jSRkbUjLGsOb9S8r",
            "created_date": "2017-11-18T18:34:30.017Z",
            "email": "cpavin@usopen.com",
            "group_id": 1,
            "name": "Corey Pavin",
            "user_id": 95,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component) />,
        <Connect(Transaction) />,
        <Connect(Comments) />,
        <Connect(UserProfile) />,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="join-group"
          to="/joingroup"
>
          SWITCH GROUPS
</NavLink>,
      ],
      "className": "user-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "SWITCH GROUPS",
          "className": "join-group",
          "to": "/joingroup",
        },
        "ref": null,
        "rendered": "SWITCH GROUPS",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component) />,
          <Connect(Transaction) />,
          <Connect(Comments) />,
          <Connect(UserProfile) />,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="join-group"
            to="/joingroup"
>
            SWITCH GROUPS
</NavLink>,
        ],
        "className": "user-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "SWITCH GROUPS",
            "className": "join-group",
            "to": "/joingroup",
          },
          "ref": null,
          "rendered": "SWITCH GROUPS",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`user tests should match the snapshot if user does not have a group 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <User
    User={
        Object {
            "authrocket_id": "usr_0vXWX9jSRkbUjLGsOb9S8r",
            "created_date": "2017-11-18T18:34:30.017Z",
            "email": "cpavin@usopen.com",
            "group_id": null,
            "name": "Corey Pavin",
            "user_id": 95,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component) />,
        <Connect(Transaction) />,
        <Connect(Comments) />,
        <Connect(UserProfile) />,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="join-group"
          to="/joingroup"
>
          JOIN A GROUP
</NavLink>,
      ],
      "className": "user-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "JOIN A GROUP",
          "className": "join-group",
          "to": "/joingroup",
        },
        "ref": null,
        "rendered": "JOIN A GROUP",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component) />,
          <Connect(Transaction) />,
          <Connect(Comments) />,
          <Connect(UserProfile) />,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="join-group"
            to="/joingroup"
>
            JOIN A GROUP
</NavLink>,
        ],
        "className": "user-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "JOIN A GROUP",
            "className": "join-group",
            "to": "/joingroup",
          },
          "ref": null,
          "rendered": "JOIN A GROUP",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
